advisors/spring-ai-advisors-vector-store/src/main/java/org/springframework/ai/chat/client/advisor/vectorstore/VectorStoreChatMemoryAdvisor.java

```diff
@@ -48,14 +49,24 @@
  * @author Christian Tzolov
  * @author Thomas Vitale
  * @author Oganes Bozoyan
+ * @author Mark Pollack
  * @since 1.0.0
  */
-public class VectorStoreChatMemoryAdvisor extends AbstractChatMemoryAdvisor<VectorStore> {
+public class VectorStoreChatMemoryAdvisor implements BaseChatMemoryAdvisor {
+
+	private static final Logger logger = LoggerFactory.getLogger(VectorStoreChatMemoryAdvisor.class);
+
+	public static final String CHAT_MEMORY_RETRIEVE_SIZE_KEY = "chat_memory_response_size";
 
 	private static final String DOCUMENT_METADATA_CONVERSATION_ID = "conversationId";
 
 	private static final String DOCUMENT_METADATA_MESSAGE_TYPE = "messageType";
 
+	/**
+	 * The default chat memory retrieve size to use when no retrieve size is provided.
+	 */
+	public static final int DEFAULT_CHAT_MEMORY_RESPONSE_SIZE = 100;
```

Comment:
this should probably be renamed to something like `DEFAULT_TOP_K` ?

Also, isn't 100 a bit too much? Should it be something lower? Like 20? (the value we use for messages)

Reply:
ha, did it already!